src/dune_lang/syntax.ml

```diff
@@ -292,10 +305,6 @@ let greatest_supported_version t =
   Option.value_exn
     (Supported_versions.greatest_supported_version t.supported_versions)
 
-let greatest_supported_version_for_dune_lang t ~dune_lang_ver =
-  Supported_versions.greatest_supported_version_for_dune_lang
```

Comment:
I guess we could also remove `Syntax.Supported_versions.greatest_supported_version_for_dune_lang` ?

Reply:
Nevermind, it is used elsewhere in this PR.